rio/src/RIO/Prelude/RIO.hs

```diff
@@ -51,67 +50,47 @@ instance PrimMonad (RIO env) where
     primitive = RIO . ReaderT . const . primitive
 
 data SomeRef a
-  = SomeRef { readSomeRef :: !(IO a)
-            , writeSomeRef :: !(a -> IO ())
-            }
+  = SomeRef !(IO a) !(a -> IO ())
+
+readSomeRef :: SomeRef a -> IO a
+readSomeRef (SomeRef x _) = x
+
+writeSomeRef :: SomeRef a -> a -> IO ()
+writeSomeRef (SomeRef _ x) = x
 
 modifySomeRef :: SomeRef a -> (a -> a) -> IO ()
 modifySomeRef (SomeRef read write) f =
   (f <$> read) >>= write
 
-{-
-
-NOTE:
-
-I believe there is still some merit to this idea, leaving for now
-to assess and discuss
-
-class ToSomeRef (ref :: * -> *) where
-  type RefConstraints ref a ::  Constraint
-  toSomeRef :: RefConstraints ref a => ref a -> SomeRef a
-
-instance ToSomeRef IORef where
-  type RefConstraints IORef a = (() :: Constraint)
-  toSomeRef = ioRefToSomeRef
-
-instance (PrimState IO ~ base) => ToSomeRef (URef base) where
-  type RefConstraints (URef base) a = Unbox a
-  toSomeRef = uRefToSomeRef
-
-runRIORef
-  :: (ToSomeRef ref, MonadIO m, RefConstraints ref a)
-  => ref a -> RIO (SomeRef a) b -> m b
-runRIORef env action = runRIO (toSomeRef env) action
-
--- ref <- newURef 0
--- runRIORef ref $ do
---   modify (+1)
--- readURef ref ==> 1
-
--}
-
+ioRefToSomeRef :: IORef a -> SomeRef a
 ioRefToSomeRef ref = do
   SomeRef (readIORef ref)
           (\val -> atomicModifyIORef' ref (\old -> (val, ())))
 
-
+uRefToSomeRef :: Unbox a => URef RealWorld a -> SomeRef a
 uRefToSomeRef ref = do
   SomeRef (readURef ref) (writeURef ref)
 
-
 instance MonadState s (RIO (SomeRef s)) where
   get = ask >>= liftIO . readSomeRef
   put st = do
     ref <- ask
     liftIO $ writeSomeRef ref st
 
-instance Monoid w => MonadWriter w (RIO (SomeRef w)) where
+instance (Show w, Monoid w) => MonadWriter w (RIO (SomeRef w)) where
```

Comment:
Why do we need the `Show` constraint here?

Reply:
That was for debugging and i forgot to remove 👍